jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2365,7 +2373,36 @@ public T newInstance() throws IllegalAccessException, InstantiationException {
 		checkNonSunProxyMemberAccess(security, callerClassLoader, Member.PUBLIC);
 	}
 
+/*[IF JAVA_SPEC_VERSION >= 12]*/
+	Class<?> callerClazz = getStackClass(1);
+	if (callerClazz.classLoader == ClassLoader.bootstrapClassLoader) {
+		/* newInstanceImpl() is required for all bootstrap classes to avoid an infinite loop
+		 * when calling copyConstructor.invoke() in cacheConstructor() at the bootstrap stage
+		 * as the constructors of bootstrap classes are not yet cached for use at that time.
+		 */
+		return (T)J9VMInternals.newInstanceImpl(this);
+	} else {
+		try {
+			Constructor<?> ctr = getDeclaredConstructor();
+			if (reflectionFactoryOffset == -1) {
+				reflectionFactoryOffset = getFieldOffset("reflectionFactory"); //$NON-NLS-1$
+			}
+			if (reflectionFactory == null) {
+				ReflectionFactory localReflectionFactory = AccessController.doPrivileged(new ReflectionFactory.GetReflectionFactoryAction());
+				writeFieldValue(reflectionFactoryOffset, localReflectionFactory);
```

Comment:
Apologies. Due to how the reflection factory is created, the original code was fine and using `writeFieldValue()` isn't required.

Reply:
Restored back to the original code with the reflection factory.